src/typechecker.ts

```diff
@@ -179,6 +185,37 @@ const getExpressionVisitor =
     }
   };
 
+type ErrorForUsingTypeXAsTypeY =
+  | null
```

Comment:
In what way does `null` represent something different from `'other'`? You should document that here in a comment.

Reply:
Done. (It's "no error".)